libsignal-service/src/lib.rs

```diff
@@ -9,3 +9,36 @@ pub const USER_AGENT: &'static str =
     concat!(env!("CARGO_PKG_NAME"), "-rs-", env!("CARGO_PKG_VERSION"));
 
 pub struct TrustStore;
+
+/// Attachment represents an attachment received from a peer
+///
+/// Source: `textsecure/textsecure.go`
+pub struct Attachment {
+    // pub R io.Reader whatever file
+    pub mime_type: String,
+}
+
+/// Group holds group metadata
+///
+/// Source: `textsecure/groups.go`
+pub struct Group {
+    pub id: Vec<u8>,  // bytes
+    pub hex_id: String,
+    pub flags: u32,
+    pub name: String,
+    pub members: Vec<String>,
+    // pub avatar: io.Reader yaml:"-"
```

Comment:
probably same story as for Attachment

Reply:
This was discussed and it will be `pub avatar: Option<Vec<u8>>,`